advanced_alchemy/exceptions.py

```diff
@@ -84,3 +88,34 @@ class IntegrityError(RepositoryError):
 
 class NotFoundError(RepositoryError):
     """An identity does not exist."""
+
+
+class MultipleResultsFoundError(AdvancedAlchemyError):
+    """A single database result was required but more than one were found."""
+
+
+@contextmanager
+def wrap_sqlalchemy_exception() -> Any:
+    """Do something within context to raise a `RepositoryError` chained
+    from an original `SQLAlchemyError`.
+
+        >>> try:
+        ...     with wrap_sqlalchemy_exception():
+        ...         raise SQLAlchemyError("Original Exception")
+        ... except RepositoryError as exc:
+        ...     print(f"caught repository exception from {type(exc.__context__)}")
+        ...
+        caught repository exception from <class 'sqlalchemy.exc.SQLAlchemyError'>
```

Comment:
is this valid? does sphinx expect `.. code-block::` directive instead?

Reply:
This hasn't been modified from what it was previously.

Reply:
it works!